### DISCUSSION TITLE
Using CustomParticle as a key in a dictionnary

### DISCUSSION THREAD

**Discussion by Quettle** [PlasmaPy/PlasmaPy#1192] [Particles] [answered]
Hi, 
I am currently integrating Plasmapy in a pic simulation to make its input user friendly and keep track of the involved species. This concept is working great with a dictionary using `Particle` objects as keys to index arrays. However this simulation has to work with molecules, such as I2 and multiple degenerated stats of some elements that requires the use of `CustomParticle` objects. Should I make a pull request with implementations of `__eq__` and `__hash__` or is there a reason for them being absent from `CustomParticle`.

Thank you for your help.
Nicolas Lequette

**Comment by rocco8773** [comment, marked as answer]
@namurphy I think you're probably the most qualified person to make a comment on this.  Any thoughts?

**Reply by namurphy** [reply]
Yeah, I think that it would be worth `CustomParticle` (and also `DimensionlessParticle`) to be able to be used as dictionary keys.  I had previously been hesitant about that because they are currently mutable, but it is in our plan to make them immutable (#1110).  In any case, please feel free to make a pull request for this.  We'll probably be releasing version 0.7.0 within the next few weeks, so this change could be included in an official release fairly soon.  Thanks!
